internal/cmd/plugin/hibernate/on.go

```diff
@@ -0,0 +1,209 @@
+			stopped, err := isInstanceStopped(ctx, instance)
+			if err != nil {
+				return err
+			}
+			if !stopped {
+				return fmt.Errorf("instance still running")
+			}
+			return nil
+		}); err != nil {
+			return err
+		}
+	}
+	contextLogger.Info("cluster fenced")
+
+	// we annotate the primary instance PVCs with the cluster data
+	pvcs, err := resources.GetInstancePVCs(ctx, clusterName, primaryInstance.Name)
+	if err != nil {
+		return err
+	}
+
+	contextLogger.Info("fetching primary pg_controldata output")
+	controlData, err := getPGControlData(ctx, primaryInstance)
+	if err != nil {
+		return err
+	}
+	contextLogger.Info("primary pg_controldata output fetched")
+
+	contextLogger.Info("annotating the PVC with the cluster manifest")
+	if err := annotatePVCs(ctx, pvcs, cluster, controlData); err != nil {
+		return err
+	}
+	contextLogger.Info("PVC annotation complete")
+
+	contextLogger.Info("destroying the primary instance while preserving the pvc")
+	serial, err := specs.GetNodeSerial(primaryInstance.ObjectMeta)
+	if err != nil {
+		return err
+	}
+
+	if err := destroy.Destroy(ctx, clusterName, strconv.Itoa(serial), true); err != nil {
+		return err
+	}
+	contextLogger.Info("primary instance destroy completed")
+
+	contextLogger.Info("deleting the cluster resource")
+	if err := plugin.Client.Delete(ctx, &cluster); err != nil {
+		return err
+	}
+	contextLogger.Info("cluster resource deletion complete")
+
+	contextLogger.Info("hibernation process completed")
+	return nil
+}
+
+func annotatePVCs(
+	ctx context.Context,
+	pvcs []corev1.PersistentVolumeClaim,
+	cluster v1.Cluster,
+	pgControlData string,
+) error {
+	for _, pvc := range pvcs {
+		if err := retry.OnError(retry.DefaultBackoff, func(err error) bool { return true }, func() error {
+			var currentPVC corev1.PersistentVolumeClaim
+			if err := plugin.Client.Get(
+				ctx,
+				types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace},
+				&currentPVC,
+			); err != nil {
+				return err
+			}
+
+			if currentPVC.Annotations == nil {
+				currentPVC.Annotations = map[string]string{}
+			}
+			origPVC := currentPVC.DeepCopy()
+
+			bytes, err := json.Marshal(&cluster)
+			if err != nil {
+				return err
+			}
+
+			currentPVC.Annotations[utils.HibernateClusterManifestAnnotationName] = string(bytes)
+			currentPVC.Annotations[utils.HibernatePgControlDataAnnotationName] = pgControlData
+
+			return plugin.Client.Patch(ctx, &currentPVC, client.MergeFrom(origPVC))
+		}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getPGControlData(ctx context.Context,
+	pod corev1.Pod,
+) (string, error) {
+	timeout := time.Second * 2
+	clientInterface := kubernetes.NewForConfigOrDie(plugin.Config)
+	stdout, _, err := utils.ExecCommand(
+		ctx,
+		clientInterface,
+		plugin.Config,
+		pod,
+		specs.PostgresContainerName,
+		&timeout,
+		"pg_controldata")
+	if err != nil {
+		return "", err
+	}
+
+	return stdout, nil
+}
+
+func isInstanceStopped(
```

Comment:
wondering if we could leverage `func (instance *Instance) isStatusRunning()` from the instance manager.
Going to experiment a bit...